test/libsolidity/semanticTests/errors/require_error_string_memory.sol

```diff
@@ -0,0 +1,15 @@
+error CustomError(string);
+
+contract C
+{
+    function f() external pure
+    {
+        string memory reason = "errorReason";
+        require(false, CustomError(reason));
```

Comment:
Would be best to have more than one `require` call in most of these semantic tests with different types. Otherwise we're not really testing whether the generated helper function is unique enough.

Reply:
Also, I think we might want to try a few more weird invalid syntax cases as suggested by Daniel earlier, to make sure that nothing really slips through the cracks. I see only one of those.

For example, what about `require` nested in `require`?

Reply:
One more thing I just remembered - we should have at least one semantic test with a type that takes more than one stack slot.